jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/list/ExtendedPagedTableTest.java

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+
+package org.jbpm.workbench.common.client.list;
+
+import java.util.List;
+
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.jbpm.workbench.common.model.GenericSummary;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.uberfire.ext.services.shared.preferences.GridGlobalPreferences;
+import org.uberfire.ext.widgets.table.client.DataGrid;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class ExtendedPagedTableTest {
+
+    @Mock
+    GridGlobalPreferences preferences;
+
+    @Mock
+    DataGrid dataGrid;
+
+    ExtendedPagedTable table;
+
+    @Before
+    public void setup() {
+        table = new ExtendedPagedTable(preferences);
+        table.dataGrid = dataGrid;
+    }
+
+    @Test
+    public void testHasSelectedItems() {
+        when(dataGrid.getVisibleItems()).thenReturn(emptyList());
+
+        assertFalse(table.hasSelectedItems());
+
+        GenericSummary gs = mock(GenericSummary.class);
+        when(gs.isSelected()).thenReturn(true,
+                                         false);
+        when(dataGrid.getVisibleItems()).thenReturn(asList(gs,
+                                                           gs));
+
+        assertTrue(table.hasSelectedItems());
+    }
+
+    @Test
+    public void testDeselectAllItems() {
+        GenericSummary gs = mock(GenericSummary.class);
+        when(dataGrid.getVisibleItems()).thenReturn(asList(gs,
+                                                           gs));
+
+        table.deselectAllItems();
+
+        verify(gs,
+               times(2)).setSelected(false);
+    }
+
+    @Test
+    public void testGetSelectedItems() {
+        GenericSummary gs = mock(GenericSummary.class);
+        when(gs.isSelected()).thenReturn(true,
+                                         false);
+        when(dataGrid.getVisibleItems()).thenReturn(asList(gs,
+                                                           gs));
+
+        List<GenericSummary> selected = table.getSelectedItems();
+
+        assertNotNull(selected);
+        assertEquals(1,
+                     selected.size());
+    }
+
+    @Test
+    public void testIsAllItemsSelected() {
+        GenericSummary gs = mock(GenericSummary.class);
+        when(gs.isSelected()).thenReturn(true,
+                                         true,
+                                         false);
+        when(dataGrid.getVisibleItems()).thenReturn(emptyList(),
+                                                    asList(gs,
+                                                           gs),
+                                                    singletonList(gs));
+
+        assertTrue(table.isAllItemsSelected());
```

Comment:
@cristianonicolai @bsremac verified, the bulk actions are enabled, at the first access to a view without results